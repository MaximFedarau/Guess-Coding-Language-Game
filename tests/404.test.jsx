import { render } from '@testing-library/react';

import Page404 from 'pages/404';

// mocking Head Next.js component to check the <title>
jest.mock('next/head', () => {
  return {
    __esModule: true,
    default: ({ children }) => {
      return <>{children}</>;
    },
  };
});

describe('UI is correct', () => {
  test('Standard and unchanged UI.', () => {
    const page = render(<Page404 />);
    // <body> + <title> check
    expect(page.baseElement).toMatchSnapshot();
  });
});
