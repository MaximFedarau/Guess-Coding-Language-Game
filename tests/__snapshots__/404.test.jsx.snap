// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI is correct. Standard and unchanged UI. 1`] = `
<body>
  <div>
    <title>
      Page not found
    </title>
    <article
      class="sc-a4449903-0 kQQPqm"
    >
      <svg
        color="#1572A1"
        fill="currentColor"
        height="16vw"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(21, 114, 161); min-height: 144px; min-width: 144px;"
        viewBox="0 0 512 512"
        width="16vw"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M160 164s1.44-33 33.54-59.46C212.6 88.83 235.49 84.28 256 84c18.73-.23 35.47 2.94 45.48 7.82C318.59 100.2 352 120.6 352 164c0 45.67-29.18 66.37-62.35 89.18S248 298.36 248 324"
          fill="none"
          stroke-linecap="round"
          stroke-miterlimit="10"
          stroke-width="40"
        />
        <circle
          cx="248"
          cy="399.99"
          r="32"
        />
      </svg>
    </article>
  </div>
</body>
`;
